contracts/vault/src/execute.rs

```diff
@@ -141,88 +144,41 @@ fn execute_redistribute_funds(deps: DepsMut, env: Env, info: MessageInfo) -> Res
         .amount;
 
     let excess = vault_balance.saturating_sub(pending);
-
     if excess.is_zero() {
         return Err(anyhow!("No excess to redistribute"));
     }
 
-    let markets: Vec<String> = state::MARKET_ALLOCATIONS
-        .keys(deps.storage, None, None, Order::Ascending)
-        .filter_map(|market_id_res| {
-            let market_id = market_id_res.ok()?;
-            let resp: StatusResp = deps
-                .querier
-                .query(&QueryRequest::Wasm(WasmQuery::Smart {
-                    contract_addr: market_id.clone(),
-                    msg: to_json_binary(&MarketQueryMsg::Status { price: None }).ok()?,
-                }))
-                .ok()?;
-            if let Token::Native { denom, .. } = &resp.collateral {
-                if denom == &config.usdc_denom {
-                    Some(Ok(market_id))
-                } else {
-                    None
-                }
-            } else {
-                None
-            }
-        })
-        .collect::<StdResult<Vec<String>>>()?;
-
-    let mut utilizations: Vec<(String, Uint128)> = markets
-        .into_iter()
-        .filter_map(|market| {
-            let resp: StatusResp = deps
-                .querier
-                .query(&QueryRequest::Wasm(WasmQuery::Smart {
-                    contract_addr: market.clone(),
-                    msg: to_json_binary(&MarketQueryMsg::Status { price: None })
-                        .expect("Serialize Market Query Msg"),
-                }))
-                .ok()?;
-
-            let utilization =
-                (resp.liquidity.total_lp + resp.liquidity.total_xlp).unwrap_or_default();
-
-            let value = Uint128::from(utilization.into_u128().expect("Error LpToken to Uint128"));
-            Some((market, value))
-        })
-        .collect();
-
-    utilizations.sort_by(|a, b| b.1.cmp(&a.1));
-
-    let total_bps: u16 = config.markets_allocation_bps.iter().sum();
+    let total_bps: u16 = config.markets_allocation_bps.values().sum();
```

Comment:
Do we also want to check that the total adds up to 100%, or at least doesn't exceed 100%?